04_containers.Rmd

```diff
@@ -663,95 +669,56 @@ shows just the beginning of the data table for an overview.
 `knitr::kable()` is for printing the information more nicely.
 
 ```{r}
-head(rowData(se))
+head(rowData(t6se))
 ```
 
-These taxonomic rank names (column names) are not real rank
-names. Let’s replace them with real rank names.
-
-In addition to that, the taxa names include, e.g., '"k__' before the name, so let's
-make them cleaner by removing them. 
+We further polish the feature names by removing unnecessary characters and then replace the original rowData with its updated version.
 
 ```{r}
-names(rowData(se)) <- c("Kingdom", "Phylum", "Class", "Order", 
-                        "Family", "Genus")
-
-# Goes through the whole DataFrame. Removes '.*[kpcofg]__' from strings, where [kpcofg] 
-# is any character from listed ones, and .* any character.
-rowdata_modified <- BiocParallel::bplapply(rowData(se), 
-                                           FUN = stringr::str_remove, 
-                                           pattern = '.*[kpcofg]__')
-
 # Genus level has additional '\"', so let's delete that also
-rowdata_modified <- BiocParallel::bplapply(rowdata_modified, 
+rowdata_modified <- BiocParallel::bplapply(rowData(tse), 
                                            FUN = stringr::str_remove, 
                                            pattern = '\"')
 
-# rowdata_modified is a list, so it is converted back to DataFrame format. 
-rowdata_modified <- DataFrame(rowdata_modified)
-
-# And then assigned back to the SE object
-rowData(se) <- rowdata_modified
+# rowdata_modified is a list, so convert this back to DataFrame format. 
+# and assign the cleaned data back to the TSE rowData
+rowData(tse) <- DataFrame(rowdata_modified)
 
 # Now we have a nicer table
-head(rowData(se))
+head(rowData(tse))
 ```
 
-We notice that the imported biom file did not contain the sample meta data
-yet, so it includes an empty data frame.
+We notice that the imported biom file did not contain any colData yet,
+so only an empty dataframe appears in this slot.
 
 ```{r}
-head(colData(se))
+head(colData(tse))
 ```
 
-Let us add a sample metadata file.
+Let us add colData from the sample metadata, which is stored in a CSV file.
 
 ```{r}
-# We use this to check what type of data it is
-# read.table(sample_meta_file_path)
-
-# It seems like a comma separated file and it does not include headers
-# Let us read it and then convert from data.frame to DataFrame
-# (required for our purposes)
-sample_meta <- DataFrame(read.table(sample_meta_file_path, sep = ",", header = FALSE))
-
-# Add sample names to rownames
-rownames(sample_meta) <- sample_meta[,1]
-
-# Delete column that included sample names
-sample_meta[,1] <- NULL
+# CSV file without headers and rownames in the first column
+sample_meta <- read.table(sample_meta_file_path,
+                          sep = ",", header = FALSE, row.names = 1)
 
 # We can add headers
 colnames(sample_meta) <- c("patient_status", "cohort", "patient_status_vs_cohort", "sample_name")
```

Comment:
Why not read the metadata file with `DataFrame(read.csv(sample_meta_file_path, sep = ",", header = TRUE))` as this will read the headers automatically (no need to add afterwards; although it must be made sure that examples will work with the original names).

Reply:
The colnames specified in line 706 are easier to understand than the original names, so I guess that's why it was done this way.

Reply:
Yes that is the reason. Adding headers is not necessary here, although it might be convenient. 
 

If we keep that, then it could be pedagogically useful to extend the example a bit and first show the "difficult" names, then explain that they are not desirable and show how the user can optionally change those initial names that were already read from the file automatically if they choose so. 

We can do that. It is about finding the optimal balance and focus on teaching the new microbiome methods vs. teaching how to use R for standard data polishing.